### PR TITLE
dmenu: add conf option

### PR DESCRIPTION
**Pull request by head-gardener**
## Description of changes

Allow users to set their own config.h file, similarly to dwm.
See [here](https://github.com/head-gardener/config/blob/a6667527d9e02eea8173b50b6ec343cbbd776084/overlays/dmenu/default.nix#L28) for a usage example.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/dmenu/default.nix [hunk 1]
```diff
@@ -13,10 +13,17 @@ stdenv.mkDerivation rec {
...
-  postPatch = ''
-    sed -ri -e 's!\<(dmenu|dmenu_path|stest)\>!'"$out/bin"'/&!g' dmenu_run
-    sed -ri -e 's!\<stest\>!'"$out/bin"'/&!g' dmenu_path
-  '';
+  postPatch =
+    let
+      # Allow users to set the config.def.h file containing the configuration
+      configFile =
+        if lib.isDerivation conf || builtins.isPath conf
+        then conf else lib.writeText "config.def.h" conf;
```
Comment (AndersonTorres) [inline pkgs/applications/misc/dmenu/default.nix:21]:
```suggestion
        if lib.isDerivation conf || builtins.isPath conf then 
          conf 
        else
          lib.writeText "config.def.h" conf;
```

Comment (head-gardener) [reply pkgs/applications/misc/dmenu/default.nix:21]: Why though? Formatter doesn't seem to care.

**Comment by head-gardener** [comment]
cc @pSub @globin

File: pkgs/applications/misc/dmenu/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/applications/misc/dmenu/default.nix:21]: OK.

**Pull request closed without merging**